### PR TITLE
fix: Correct P0 deprecation

### PR DESCRIPTION
**Pull request by jrwrigh**
Misuse of `PetscOptionsDeprecated` caused the `P0` option to be ignored for some reason. 

TODO:
- [ ] Correctly use `PetscOptionsDeprecated` for the symmetry boundary conditions
- [ ] Add test that modifies the `P0`/`pressure_infinity` flag. That will (hopefully) help catch these issues more easily.

### CODE REVIEW

File: examples/fluids/index.md [hunk 1]
```diff
@@ -855,20 +855,17 @@ numerous terms in the STG formulation.
 
 #### Internal Damping Layer (IDL)
 The STG inflow boundary condition creates large amplitude acoustic waves.
-We use an internal damping layer (IDL) to damp them out without disrupting the synthetic structures developing into natural turbulent structures. This implementation was inspired from
-{cite}`shurSTG2014`, but is implemented here as a ramped volumetric forcing
-term, similar to a sponge layer (see 8.4.2.4 in {cite}`colonius2023turbBC` for example). It takes the following form:
+We use an internal damping layer (IDL) to damp them out without disrupting the synthetic structures developing into natural turbulent structures.
+This implementation was inspired from {cite}`shurSTG2014`, but is implemented here as a ramped volumetric forcing term, similar to a sponge layer (see 8.4.2.4 in {cite}`colonius2023turbBC` for example).
```
Comment (jeremylt) [inline examples/fluids/index.md:859]:
```suggestion
This implementation was inspired by {cite}`shurSTG2014`, but is implemented here as a ramped volumetric forcing term, similar to a sponge layer (see 8.4.2.4 in {cite}`colonius2023turbBC` for example).
```

perhaps?

**Review by jeremylt — APPROVED** [review]
